express/src/controllers/sign-in.ts

```diff
@@ -26,16 +29,45 @@ export const showLoginOtpMobile = async function (req: Request, res: Response) {
             formActionUrl: "/sign-in-otp-mobile",
             active: "sign-in",
             textMessageNotReceivedUrl: "/resend-text-code"
-
         });
     } else {
         res.redirect('/sign-in');
     }
 }
 
-export const processLoginOtpMobile = async function (req: Request, res: Response) {
-    // TO DO add the functionality to process the login mobile otp
-    if (true) { // because OTP was correct and we've implemented that
+export const processLoginOtpMobile = async function(req: Request, res: Response) {
+
+    if(!req.session.cognitoUser?.Username || !req.body['sms-otp'] || !req.session.session) {
+        res.render('there-is-a-problem.njk');
+        return;
+    }
+
+    const cognitoClient: CognitoInterface = req.app.get('cognitoClient');
+    let response!: AdminRespondToAuthChallengeCommandOutput;
+    try {
+        response = await cognitoClient.respondToMfaChallenge(req.session.cognitoUser?.Username as string, req.body['sms-otp'], req.session.session as string);
+    } catch (error) {
+        if(error instanceof CodeMismatchException) {
+            const errorMessages: Map<string, string> = new Map<string, string>();
+            errorMessages.set('smsOtp', 'The code you entered is not correct or has expired - enter it again or request a new code')
+            res.render('common/check-mobile.njk', {
+                mobileNumber: req.session.mobileNumber,
+                formActionUrl: "/sign-in-otp-mobile",
+                errorMessages: errorMessages
+            })
+        }
+    }
+    req.session.authenticationResult = response.AuthenticationResult;
+
+    const payload = (req.session.authenticationResult?.IdToken as string).split('.');
```

Comment:
What is `authenticationResult` is null? We would get a null pointer exception on the next line, wouldn't we?